groups/bdl/bdlb/bdlb_pcgrandomgenerator.h

```diff
@@ -0,0 +1,297 @@
+//      bsl::uint32_t result;
+//
+//      do {
+//          result = d_pcg.generate() & 7;
+//      } while (result > 5);
+//
+//      return result + 1;
+//  }
+//..
+// Now, we can use our 'Dice' class to get the random numbers needed to
+// simulate a game of craps.  Note that the game of craps requires two dice.
+//
+// We can instantiate a single 'Die' and role it twice,
+//..
+//  void rollOneDieTwice()
+//  {
+//      Die a(123, 456);
+//
+//      int d1 = a.roll();
+//      int d2 = a.roll();
+//
+//      cout << "d1 = " << d1 << ", d2 = " << d2 << endl;  // d1 = 3, d2 = 5
+//  }
+//..
+// Alternatively, we could create two instances of 'Die', with separate initial
+// seeds, and role each one once:
+//..
+//  void rollTwoDice()
+//  {
+//      Die a(123, 123);
+//      Die b(456, 456);
+//
+//      int d1 = a.roll();
+//      int d2 = b.roll();
+//
+//      cout << "d1 = " << d1 << ", d2 = " << d2 << endl;  // d1 = 3, d2 = 1
+//  }
+//..
+// Note that the specification of separate seeds is important to produce a
+// proper distribution for our game.  If we had shared the seed value each die
+// would always produce the same sequence of values as the other.
+//..
+//  void shareSeed()
+//  {
+//      Die a(123, 456);  // BAD IDEA
+//      Die a(123, 456);  // BAD IDEA
+//
+//      int d1 = a.roll();
+//      int d2 = b.roll();
+//      assert(d2 == d1);
+//
+//  }
+//..
+///Example 2: Generating a random 32-bit number
+///- - - - - - - - - - - - - - - - - - - - - - -
+// The PCGRandomGenerator constructor takes two 64-bit constants (the initial
+// state, and the rng sequence selector.  Instances of 'PCGRandomGenerator' with
+// different sequence selectors will never have random sequences that coincide.
+// In order to obtain nondeterministic output for each instance, on a Unix
+// system random bytes can be obtained from '/dev/urandom'.  The component 
+// bdlb_randomdevice provides such functionality.  Also illustrated below is a 
+// a backup mechanism for obtaining random bytes.
+//..
+//  uint64_t state;
+//  if (0 != RandomDevice::getRandomBytes((unsigned char *)&state,
+//                                                         sizeof(state))) {
+//          state = time(NULL) ^ (intptr_t)&bsl::printf;  // fallback state
+//  }
+//  uint64_t streamSelector;
+//  if (0 != RandomDevice::getRandomBytes((unsigned char *)&streamSelector,
+//                                              sizeof(streamSelector))) {
+//           streamSelector = time(NULL) ^
+//                   (intptr_t)&bsl::printf;  // fallback streamSelector
+//  }
+//  bdlb::PCGRandomGenerator  rng(state, streamSelector);
+//  bsl::uint32_t randomInt = rng.generate();
+//..
+
+#include <bsl_cstdint.h>
+
+namespace BloombergLP {
+namespace bdlb {
+
+                                 // ========================
+                                 // class PCGRandomGenerator
+                                 // ========================
+class PCGRandomGenerator {
```

Comment:
I think, based on the preponderance of similar abbreviations in BDE (e.g. `bdlt::Iso8601`, `bdlt::DateTz`, `bdlde::Md5`) that this should be `PcgRandomGenerator`